src/gt4py/next/iterator/ir_utils/ir_makers.py

```diff
@@ -389,3 +396,34 @@ def _impl(*its: itir.Expr) -> itir.FunCall:
 def map_(op):
     """Create a `map_` call."""
     return call(call("map_")(op))
+
+
+def cartesian_domain(ranges: Dict[str, Tuple[Any, Any]], out_field: str,
```

Comment:
This function looks unnecessarily complicated, just a dict with start end stop of the ranges should be sufficient. This would be a good doctest to add.
```
>>> cartesian_domain({"IDim": (0, 10), "JDim": (0, 20)})
c⟨ IDim: [0, 10), JDim: [0, 20) ⟩
```